src/cmd/ksh93/tests/functions.sh

```diff
@@ -1230,4 +1230,26 @@ foo()
 
 $SHELL -c 'function ftest { ftest2; }; function ftest2 { unset -f ftest; }; ftest' 2> /dev/null || err_exit 'unset of function in the calling stack fails'
 
+
+# When calling a function inside of command substitution and the function 
+# both opens and closes the file descriptor "3", the output generated by 
+# the function is not returned as a result of the command substitution.
+function run_in_subshell
+{
+    # Create a temporary file
+    tmpfile=$(mktemp)
+    ls > $tmpfile
+
+    # Bug is not exhibited if choosing a higher-numbered file
+    # descriptor (e.g. 4).
```

Comment:
I'll bet $20 this statement is wrong. It's only true if there are any unused (i.e., closed) file descriptors between 2 and the "higher-numbered file descriptor". That's because all UNIX implementations, going back to the very first, guarantee that doing something that returns a file descriptor (e,g., calling `open()`) returns the lowest numbered unused file descriptor. So you're less likely to notice this problem if the function uses a fd other than 3.

Reply:
I'm all for incorporating fixes from other distros (e.g., Fedora) but I would like to avoid including bogus ideas like this.